platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/cli/EventStreamSingleFileRepairTest.java

```diff
@@ -0,0 +1,263 @@
+ */
+class EventStreamSingleFileRepairTest {
+
+    /**
+     * The temporary directory to use for the unit test.
+     */
+    @TempDir
+    Path tmpDir;
+
+    /**
+     * Generates events, feeds them to consensus, then writes these consensus events to stream files. Once the files are
+     * written, it picks the last file, attempts to repair it with no effect, truncates the file, and then repairs it.
+     */
+    @Test
+    void repairFileTest() throws IOException, ConstructableRegistryException {
+
+        createEventStreamFiles();
+
+        // get last file in the event stream and prepare for running repair tests.
+        Path pathToLastFile;
+        try (final Stream<Path> walk = Files.walk(tmpDir)) {
+            pathToLastFile = walk.filter(path -> path.toString().endsWith(".evts"))
+                    .reduce(((path, path2) -> path2.compareTo(path) < 0 ? path : path2))
+                    .get();
+        }
+        assertNotNull(pathToLastFile, "Path was not provided to the last file in the event stream.");
+        final File lastFile = pathToLastFile.toFile();
+        final File backupFile = Path.of(pathToLastFile + ".backup").toFile();
+        final File repairedFile = Path.of(pathToLastFile + REPAIRED_SUFFIX).toFile();
+        final File damagedFile = Path.of(pathToLastFile + DAMAGED_SUFFIX).toFile();
+
+        setupAndAssertCleanStateForUnitTest(lastFile, backupFile, repairedFile, damagedFile);
+        assertFileDamage(lastFile, false);
+        final EventStreamSingleFileRepairer repairer1 = attemptRepairOfFile(lastFile, false);
+        assertBehaviorOnGoodEventStreamFile(lastFile, backupFile, repairedFile, damagedFile, repairer1);
+        assertFileDamage(lastFile, false);
+
+        damageTheLastFile(lastFile);
+        assertFileDamage(lastFile, true);
+        final EventStreamSingleFileRepairer repairer2 = attemptRepairOfFile(lastFile, true);
+        assertBehaviorOnDamagedEventStreamFile(lastFile, repairedFile, damagedFile, repairer2);
+        assertFileDamage(lastFile, false);
+    }
+
+    /**
+     * Records a new event stream to files on disk.
+     */
+    private void createEventStreamFiles() throws ConstructableRegistryException {
+        final Random random = RandomUtils.getRandomPrintSeed();
+        final int numNodes = 10;
+        final int numEvents = 100_000;
+        final Duration eventStreamWindowSize = Duration.ofSeconds(1);
+
+        // setup
+        ConstructableRegistry.getInstance().registerConstructables("com.swirlds");
+
+        // generate consensus events
+        final Deque<ConsensusRound> rounds =
+                GenerateConsensus.generateConsensusRounds(numNodes, numEvents, random.nextLong());
+        if (rounds.isEmpty()) {
+            Assertions.fail("events are excepted to reach consensus");
+        }
+        // get consensus info
+        final long roundToReportFrom = rounds.size() / 2;
+        final int numConsensusEvents = rounds.stream()
```

Comment:
`numConsensusEvents` and `lastEventTime` are unused

Reply:
While they are unused, they are return values from methods.  Is there any value in being able to set break points and see those values in a debugger?   If not, I will delete them.

Reply:
I would personally delete (and re add them if I ever needed a breakpoint). But I'm fine either way if you prefer them in 👍